tests/backtrace.rs

```diff
@@ -0,0 +1,187 @@
+use piccolo::{error::BacktraceFrame, Closure, Executor, ExternError, Lua};
+use std::collections::HashMap;
+
+#[test]
+fn test_backtrace() {
+    const SCRIPT: &str = include_str!("test_backtrace.lua");
+
+    let mut lua = Lua::full();
+
+    let executor = lua
+        .try_enter(|ctx| {
+            let closure = Closure::load(ctx, Some("test_backtrace.lua"), SCRIPT.as_bytes())?;
+            Ok(ctx.stash(Executor::start(ctx, closure.into(), ())))
+        })
+        .unwrap();
+
+    lua.finish(&executor).unwrap();
+    let result = lua.try_enter(|ctx| ctx.fetch(&executor).take_result::<()>(ctx)?);
+
+    let mut source_map = std::collections::HashMap::new();
+    source_map.insert("test_backtrace.lua".to_string(), SCRIPT.to_string());
+
+    if let Err(
+        ref err @ ExternError::Lua {
+            ref error,
+            ref backtrace,
+            ..
+        },
+    ) = result
+    {
```

Comment:
These might be easier to structure as
```rust
let Err(...) = result else {
    panic!("message");
};
```
mainly to keep the panic closer to the source